crates/shared/src/balancer_sor_api.rs

```diff
@@ -0,0 +1,288 @@
+//! Module for interacting with the Balancer SOR HTTP API.
+
```

Comment:
Maybe add a link to the docs explaining the API?
```suggestion
//! https://dev.balancer.fi/resources/smart-order-router
```